### PR TITLE
Autocomplete for FLS and Field anonymization

### PR DESCRIPTION
**Pull request by sergibondarenko**
This closes https://floragunn.atlassian.net/browse/ITT-895

To check this PR
1. Go to Create Role or Update Role page
2. Go to Index Permissions tab
3. Create Index Patterns item or unfold existing one
4. Add some index patterns to match existing indices to Index Patterns
5. Scroll to Field Level Security section
6. Click on any of the two ComboBoxes in the section
7. See a list of fields you can find in indices you matched in Index Patterns
8. Click on a field. See the field accepted
9. Add a custom field which doesn't exist in the list. See the field accepted
10. Scroll up to Index Patterns
11. Substitute the existing patterns by some patterns to match not existent indices
12. Scroll down to the Field Level Security section
13. Click on any of the two ComboBoxes in the section
14. See the message: "There aren't any options available" 

Screenshot
Screenshot 2019-07-08 at 18 50 48
Screenshot 2019-07-08 at 18 59 37

### CODE REVIEW

File: lib/configuration/backend/searchguard_configuration_plugin.js [hunk 1]
```diff
@@ -152,4 +152,17 @@ export default function (Client, config, components) {
         }
     });
 
+  Client.prototype.searchguard.prototype.getIndexMappings = ca({
```
Comment (mgustafsson1) [inline lib/configuration/backend/searchguard_configuration_plugin.js]: Heads-up: We need to change the syntax for 7.2, as discussed.

File: public/apps/configuration-react/services/ElasticsearchService.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { API } from '../utils/constants';
+
+class ElasticsearchService {
```
Comment (mgustafsson1) [inline public/apps/configuration-react/services/ElasticsearchService.js]: Heads-up: In 7.2, there is a new elasticsearch_service.ts, which exposes an ElasticsearchService.
I was a bit confused about that at first, but I think it should be clear by looking at the import that the two are unrelated...

File: public/apps/configuration-react/pages/ActionGroups/ActionGroups.js [hunk 3]
```diff
@@ -33,14 +33,14 @@ import {
...
-import { AppCacheService } from '../../services';
+import { LocalStorageService } from '../../services';
 
 class ActionGroups extends Component {
   constructor(props) {
     super(props);
 
     this.backendService = this.props.actionGroupsService;
-    this.appCache = new AppCacheService();
+    this.appCache = new LocalStorageService();
```
Comment (mgustafsson1) [inline public/apps/configuration-react/pages/ActionGroups/ActionGroups.js]: Nitpick: Since you renamed the service, maybe it would make sense to rename the variable too?
But not that important.

Comment (sergibondarenko) [reply public/apps/configuration-react/pages/ActionGroups/ActionGroups.js]: Renamed
